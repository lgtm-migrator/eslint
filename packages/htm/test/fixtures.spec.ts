import path from 'path'

import { ESLint } from 'eslint'

import * as ESLintPluginHtm from 'eslint-plugin-htm'

const eslint = new ESLint({
  useEslintrc: false,
  plugins: {
    htm: ESLintPluginHtm,
  },
  overrideConfig: {
    overrides: [
      {
        files: '*.html',
        parser: 'eslint-plugin-htm',
        plugins: ['htm'],
        rules: {
          'htm/html': 2,
        },
      },
    ],
  },
})

test('fixtures', async () => {
  const results = await eslint.lintFiles(
    path.resolve(__dirname, './fixtures/*.html'),
  )
  for (const { filePath, messages } of results) {
    expect(messages).toMatchSnapshot(path.basename(filePath))
  }
})
