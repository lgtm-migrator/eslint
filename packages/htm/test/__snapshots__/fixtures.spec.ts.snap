// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures: basic.html 1`] = `
Array [
  Object {
    "column": 2,
    "line": 1,
    "message": "Doctype must be declared first.",
    "nodeType": null,
    "ruleId": "htm/html",
    "severity": 2,
  },
]
`;
